src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java

```diff
@@ -213,10 +223,31 @@ private void onGetAnomalyDetectorResponse(GetResponse response) {
         try (XContentParser parser = RestHandlerUtils.createXContentParserFromRegistry(xContentRegistry, response.getSourceAsBytesRef())) {
             ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
             AnomalyDetector existingDetector = AnomalyDetector.parse(parser, response.getId(), response.getVersion());
-            if (!hasCategoryField(existingDetector) && hasCategoryField(this.anomalyDetector)) {
-                validateAgainstExistingMultiEntityAnomalyDetector(detectorId);
+            // We have separate flows for realtime and historical detector currently. User
+            // can't change detector from realtime to historical, vice versa.
+            if (!(existingDetector.isRealTimeDetector() ^ !anomalyDetector.isRealTimeDetector())) {
```

Comment:
The logic is somehow complicated to me. could this `if(existingDetector.isRealTimeDetector() != anomalyDetector.isRealTimeDetector()` work?

Reply:
Sure, forgot to clean this up